### DISCUSSION TITLE
Built an open-source calibration & safety middleware for Laya (and Jev)

### DISCUSSION THREAD

**Discussion by ashp15205** [NandhaKishorM/laya#341] [Show and tell]
Hey everyone! I absolutely love Laya's latency for backend routing. However, as the ecosystem shifts toward fast System 1 models (like Laya and TypeSafe Jev), I noticed they share a critical architecture flaw in production: raw, uncalibrated temperature logits can lead to dangerously overconfident routing.

To solve this, I built an open-source middleware called `sys1-decision-guard`: https://github.com/ashp15205/decision-guard

It sits on top of your Laya (or Jev) implementation and does two things:
1. **Mathematical Calibration:** It automatically applies Platt scaling to Laya's raw logits. This ensures a 100% confidence score is actually grounded in real accuracy, allowing you to safely route low-confidence edge cases to a human.
2. **Safety Guardrails:** It includes an ultra-fast Regex scanner to instantly catch prompt injections *before* they ever hit the Laya adapter.

It's completely open-source, backend-agnostic (supports both ConvAI Laya and TypeSafe Jev out of the box), and lightweight. 

If anyone here is trying to deploy Laya into a production routing environment, I'd love your feedback! 
**Repo:** https://github.com/ashp15205/decision-guard

```mermaid
flowchart TD
    A[Incoming User Request] --> B{Safety Scanner}
    
    B -- "Malicious (Regex Match)" --> C[Block & Flag]
    B -- "Safe" --> D[Model Adapter]
    
    D --> E[Laya / Jev Backend]
    E -- "Raw Logit (e.g. 100%)" --> F[Calibration Math]
    
    F -- "Platt Scaling Applied" --> G{True Confidence}
    
    G -- "High (e.g. 92%)" --> H[Auto-Route Decision]
    G -- "Low (e.g. 11%)" --> I[Human Fallback Queue]

    classDef safe fill:#064E3B,stroke:#10B981,color:#fff
    classDef warning fill:#78350F,stroke:#F59E0B,color:#fff
    classDef danger fill:#7F1D1D,stroke:#EF4444,color:#fff
    
    class A,D,E,F default
    class B,G warning
    class C,I danger
    class H safe
```
